source/common/tcp_proxy/upstream.cc

```diff
@@ -274,15 +375,116 @@ void HttpConnPool::onGenericPoolReady(Upstream::HostDescriptionConstSharedPtr& h
   callbacks_->onGenericPoolReady(nullptr, std::move(upstream_), host, address_provider, ssl_info);
 }
 
-Http2Upstream::Http2Upstream(Tcp::ConnectionPool::UpstreamCallbacks& callbacks,
-                             const TunnelingConfigHelper& config,
+CombinedUpstream::CombinedUpstream(HttpConnPool& http_conn_pool,
```

Comment:
hah I was going to suggest combining as part of the PR.  I think we should split this out as well, and land ahead of the big one, to make it clear what's functional change and what's refactor

Reply:
you mean a separate PR for changes in the existing upstreams i.e `HttpUpstream`, `Http1Upstream` and `Http2Upstream` and changes only to add new CombinedUpstream in this PR?

>suggest combining as part of the PR

you mean combining these changes with the changes for adding runtime guard to handle CONNECT with 2xx beyond 200?

Reply:
sorry I meant I was going to suggest combining the H1 and H2 upstreams.  I think this should be split out into one Pr and the 2xx into another (small) one.

Reply:
didn't we split this out so we didn't have a ton of changes in this PR?